Source/Diagnostics/FullDiagnostics.cpp

```diff
@@ -136,11 +192,27 @@ FullDiagnostics::Flush ( int i_buffer, bool /* force_flush */ )
     // is supported for BackTransformed Diagnostics, in BTDiagnostics class.
     auto & warpx = WarpX::GetInstance();
 
+    if (m_diag_type == DiagTypes::TimeAveraged) {
+        if (m_time_average_type == TimeAverageType::Static || m_time_average_type == TimeAverageType::Dynamic) {
+            // Loop over the output levels and divide by the number of steps in the averaging period
+            for (int lev = 0; lev < nlev_output; ++lev) {
+                m_sum_mf_output.at(i_buffer).at(lev).mult(1./m_average_period_steps);
```

Comment:
I wonder if this could fix the clang-tidy error "narrowing conversion from 'double' to 'amrex::Real' (aka 'float') [bugprone-narrowing-conversions,-warnings-as-errors]":
```suggestion
                m_sum_mf_output.at(i_buffer).at(lev).mult(1._rt/static_cast<amrex::Real>(m_average_period_steps));
```